src/sql/platform/accounts/common/accountStore.ts

```diff
@@ -64,13 +64,28 @@ export default class AccountStore implements IAccountStore {
 				// Remove old accounts that are now deprecated
 				try {
 					accounts = accounts.filter(account => {
+						const accountKey = account?.key?.accountId;
 						const providerKey = account?.key?.providerId;
-						// Account has no provider, remove it.
-						if (providerKey === undefined) {
+
+						// eslint-disable-next-line eqeqeq
```

Comment:
Why are you not using ===?

Reply:
== checks for null and undefined

Reply:
I prefer `typeof(accountKey) === 'undefined'` if you're going to check for that, but why is null not a valid value to filter in this case?

Reply:
Wait - I fail at reading comprehension. Ok then why would null be expected? Guess that part is unusual to me too.

And then I'd suggest using the isUndefinedOrNull method if you're going to do this check https://github.com/microsoft/azuredatastudio/blob/main/src/vs/base/common/types.ts#L77